docs/specifications/tests/Zone-TP/zone01.md

```diff
@@ -3,43 +3,262 @@
+   1. If the SOA MNAME name server name is not part of the
+      *Child Nameservers* set, then output *[Z01_MNAME_NOT_IN_NS_LIST]* with
+      SOA MNAME name server name.
+   2. Do a [DNS Lookup] of SOA MNAME name server name (A and AAAA) and add the
+      SOA MNAME name server IP address(es) found to the *MNAME Nameservers* set
+      for the same entry as the SOA MNAME name server name.
+   3. If at least one IP address from the previous step was found, then:
+      1. For each SOA MNAME name server IP for the SOA MNAME name server name do: 
+         1. If the IP address is a localhost address (127.0.0.1 or ::1), then output
+            *[Z01_MNAME_HAS_LOCALHOST_ADDR]* with SOA MNAME name server name and IP.
+         2. Else, send *SOA Query* to the name server and collect the response.
+            1. If there is a DNS response, with [RCODE Name] "NoError" and
+               an SOA record in the answer section, then:
+               1. If the AA flag is not set, then output *[Z01_MNAME_NOT_AUTHORITATIVE]* 
+                  with SOA MNAME name server name and IP address, and name server(s) IP
+                  address(es) from the *SERIAL Nameservers* set.
+               2. Else, add the SOA SERIAL field to the *MNAME Nameservers* set
+                  for the SOA MNAME name server name and IP pair.
+            2. Else if [RCODE Name] is not "NoError", then output
+               *[Z01_MNAME_UNEXPECTED_RCODE]* with [RCODE Name], SOA MNAME name server name and
+               IP address, and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+            3. Else if there is no SOA record in the answer section, then output
+               *[Z01_MNAME_MISSING_SOA_RECORD]* with SOA MNAME name server name and IP
+               address, and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+            4. Else, output *[Z01_MNAME_NO_RESPONSE]* with SOA MNAME name server name and IP
+               address, and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+         3. Go to next SOA MNAME name server IP.
+   4. Else, output *[Z01_MNAME_NOT_RESOLVE]* with SOA MNAME name server name,
+      and name server(s) IP address(es) from the *SERIAL Nameservers* set.
+   5. Go to next SOA MNAME name server name.
+
+10. If there is no SOA SERIAL in the *MNAME Nameservers* set, then
+    terminate these procedures.
+
+11. For each SOA SERIAL in *MNAME Nameservers* do:
+    1. For each SOA SERIAL in *SERIAL Nameservers* do:
+       1. Compare both SOA SERIAL values (using the arithmetic in [RFC1982]).
+          1. If the one from *SERIAL Nameservers* is higher, then add SOA MNAME name
+             server name and IP address to the *MNAME Not Master* set, and go to next
+             SOA SERIAL (from the *MNAME Nameservers* set).
+          2. Go to next SOA SERIAL.
+    2. Add SOA MNAME name server name and IP address to the *MNAME Is Master* set.
```

Comment:
Maybe it should be clarified in 11.1.2 that SOA MNAME here comes from the MNAME Nameservers set, and that it might be multiple name server names and multiple IP addresses per name.

Reply:
I'm not sure it is needed, as the other set in this part of the procedure (SERIAL Nameservers) does not contain any SOA MNAME data anyway. But sure I'll add it.

> 2. Name server IP address, SOA SERIAL ("SERIAL Nameservers")